src/components/MoneyReportHeader.tsx

```diff
@@ -107,19 +88,18 @@ function MoneyReportHeader({session, personalDetails, policy, chatReport, nextSt
     const shouldShowSettlementButton = shouldShowPayButton || shouldShowApproveButton;
     const shouldShowSubmitButton = isDraft && reimbursableTotal !== 0;
     const isFromPaidPolicy = policyType === CONST.POLICY.TYPE.TEAM || policyType === CONST.POLICY.TYPE.CORPORATE;
-    const shouldShowNextStep = isFromPaidPolicy && nextStep && !_.isEmpty(nextStep.message);
+    const shouldShowNextStep = isFromPaidPolicy && !!nextStep?.message?.length;
     const shouldShowAnyButton = shouldShowSettlementButton || shouldShowApproveButton || shouldShowSubmitButton || shouldShowNextStep;
     const bankAccountRoute = ReportUtils.getBankAccountRoute(chatReport);
     const formattedAmount = CurrencyUtils.convertToDisplayString(reimbursableTotal, moneyRequestReport.currency);
     const isMoreContentShown = shouldShowNextStep || (shouldShowAnyButton && isSmallScreenWidth);
 
     // The submit button should be success green colour only if the user is submitter and the policy does not have Scheduled Submit turned on
     const isWaitingForSubmissionFromCurrentUser = useMemo(
-        () => chatReport.isOwnPolicyExpenseChat && !policy.isHarvestingEnabled,
-        [chatReport.isOwnPolicyExpenseChat, policy.isHarvestingEnabled],
+        () => chatReport?.isOwnPolicyExpenseChat && !policy.isHarvestingEnabled,
+        [chatReport?.isOwnPolicyExpenseChat, policy.isHarvestingEnabled],
     );
-
-    const threeDotsMenuItems = [HeaderUtils.getPinMenuItem(moneyRequestReport)];
+    const threeDotsMenuItems: Array<ThreeDotsMenuItems | undefined> = [HeaderUtils.getPinMenuItem(moneyRequestReport)];
```

Comment:
Having `undefined` in the menu items doesn't look right to me, maybe we can update `HeaderUtils.getPinMenuItem` function to be
```
function getPinMenuItem(report: OnyxReport): MenuItem {
    const isPinned = !!report.isPinned;

    return {
        icon: Expensicons.Pin,
        text: Localize.translateLocal(isPinned ? 'common.unPin' : 'common.pin'),
        onSelected: Session.checkIfActionIsAllowed(() => Report.togglePinnedState(report.reportID, isPinned)),
    };
}
```
But we need to test that it works as expected and get @blazejkustra approval here since that function was migrated by him initially.

Reply:
Hi @blazejkustra , What are your views on this?

Reply:
I agree with @VickyStash 😄